File: py/1221.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        res = 0
        count = 0
        for i in range(0,len(s)):
            if s[i]=='R':
                count+=1
            else:
                count-=1
            if count==0:
                res+=1
        return res
        
                
            

File: cpp/1221.cpp
class Solution {
public:
    int balancedStringSplit(string s) {
        int count = 0;
        int res = 0;
        for (int i =0 ;i<s.size();i++){
            count+=s[i]=='R'? 1:-1;
            if (count==0){
                res+=1;
            }
        }
        return res;
    }
};
